MovieQuiz/Presentation/MovieQuizPresenter.swift

```diff
@@ -0,0 +1,150 @@
+    private let questionsAmount = 10
+    
+    private var currentQuestion: QuizQuestion?
+    private var questionFactory: QuestionFactoryProtocol?
+    private var correctAnswers = 0
+    
+    private weak var viewController: MovieQuizViewControllerProtocol?
+    
+    private var currentQuestionIndex: Int = 0
+    private let statisticService: StatisticService!
+    
+    init(viewController: MovieQuizViewControllerProtocol) {
+        self.viewController = viewController
+        
+        statisticService = StatisticServiceImpementation()
+        
+        questionFactory = QuestionFactory(moviesLoader: MoviesLoader(), delegate: self)
+        questionFactory?.loadData()
+        viewController.showLoadingIndicator()
+    }
+    
+    // MARK: Common and view interaction logic
+    
+    private func isLastQuestion() -> Bool {
+        currentQuestionIndex == questionsAmount - 1
+    }
+    
+    func restartGame() {
+        currentQuestionIndex = 0
+        correctAnswers = 0
+        questionFactory?.requestNextQuestion()
+    }
+    
+    private func switchToNextQuestion() {
+        currentQuestionIndex += 1
+    }
+    
+    func yesButtonClicked() {
+        didAnswer(isYes: true)
+    }
+    
+    func noButtonClicked() {
+        didAnswer(isYes: false)
+    }
+    
+    private func proceedToNextQuestionOrResults() {
+        if isLastQuestion() {
+            self.viewController?.show(quiz: QuizResultsViewModel(title: "Этот раунд окончен!",
+                                            text: makeResultsMessage(),
+                                            buttonText: "Сыграть ещё раз"))
+        } else {
+            switchToNextQuestion()
+            questionFactory?.requestNextQuestion()
+        }
+    }
+    
+    private func proceedWithAnswer(isCorrect: Bool) {
+        
+        didAnswer(isCorrectAnswer: isCorrect)
+        
+        viewController?.highlightImageBorder(isCorrectAnswer: isCorrect)
+        
+        DispatchQueue.main.asyncAfter(deadline: .now() + 1.0) { [weak self] in
+            guard let self = self else {
+                return
+            }
+            self.proceedToNextQuestionOrResults()
+            self.viewController?.resetImageBorderHighlighting()
+        }
+    }
+    
+    // MARK: Data transformation
+    
+    func convert(model: QuizQuestion) -> QuizStepViewModel {
+        return QuizStepViewModel(
+            image: UIImage(data: model.image) ?? UIImage(),
+            question: model.text,
+            questionNumber: "\(currentQuestionIndex + 1)/\(questionsAmount)"
+        )
+    }
+    
+    private func makeResultsMessage() -> String {
+        statisticService.store(correct: correctAnswers, total: questionsAmount)
+        
+        let bestGame = statisticService.bestGame
+        
+        let totalPlaysCountLine = "Количество сыгранных квизов: \(statisticService.gamesCount)"
+        let currentGameResultLine = "Ваш результат: \(correctAnswers)\\\(questionsAmount)"
```

Comment:
Сейчас алерт выводит значения как `6\10`, хотя дизайн требует другого наклоенния слеша
```suggestion
        let currentGameResultLine = "Ваш результат: \(correctAnswers)\/\(questionsAmount)"
```

Reply:
Спасибо, поправил.